src/fxcorrelator.py

```diff
@@ -193,6 +193,130 @@ def initialise(self, program=True, tvg=False, fake_digitiser=False):
+        sensor.set(time.time(), Sensor.NOMINAL if result else Sensor.ERROR, result)
+        self.logger.info('_sensor_feng_rx ran on %s' % (host_name))
+        Timer(10, self._sensor_feng_rx, sensor).start()
+
+    def _sensor_xeng_tx(self, sensor):
+        """
+        x-engine tx counters
+        :param sensor:
+        :return:
+        """
+        host_name = sensor.name.split('_')[2]
+        result = self.xhosts[host_name].tengbes.tx_okay()
+        sensor.set(time.time(), Sensor.NOMINAL if result else Sensor.ERROR, result)
+        self.logger.info('_sensor_xeng_tx ran on %s' % (host_name))
+        Timer(10, self._sensor_xeng_tx, sensor).start()
+
+    def _sensor_xeng_rx(self, sensor):
+        """
+        x-engine rx counters
+        :param sensor:
+        :return:
+        """
+        host_name = sensor.name.split('_')[2]
+        result = self.xhosts[host_name].check_rx_reorder()
+        sensor.set(time.time(), Sensor.NOMINAL if result else Sensor.ERROR, result)
+        self.logger.info('_sensor_xeng_rx ran on %s' % (host_name))
+        Timer(10, self._sensor_xeng_rx, sensor).start()
+
+    def _sensor_feng_phy(self, sensor):
+        """
+        f-engine PHY counters
+        :param sensor:
+        :return:
+        """
+        host_name = sensor.name.split('_')[2]
+        result = self.fhosts[host_name].check_phy_counter()
+        sensor.set(time.time(), Sensor.NOMINAL if result else Sensor.ERROR, result)
+        self.logger.info('_sensor_feng_phy ran on %s' % host_name)
+        Timer(10, self._sensor_feng_phy, sensor).start()
+        return
+
+    def _sensor_xeng_phy(self, sensor):
+        """
+        x-engine PHY counters
+        :param sensor:
+        :return:
+        """
+        host_name = sensor.name.split('_')[2]
+        result = self.xhosts[host_name].check_phy_counter()
+        sensor.set(time.time(), Sensor.NOMINAL if result else Sensor.ERROR, result)
+        self.logger.info('_sensor_xeng_phy ran on %s' % host_name)
+        Timer(10, self._sensor_xeng_phy, sensor).start()
+        return
+
+    def _xeng_qdr_okay(self, sensor):
+        """
+        x-engine QDR check
+        :param sensor:
+        :return:
+        """
+        host_name = sensor.name.split('_')[2]
+        result = self.xhosts[host_name].qdr_okay()
+        sensor.set(time.time(), Sensor.NOMINAL if result else Sensor.ERROR, result)
+        self.logger.info('_xeng_qdr_okay ran on %s' % host_name)
+        Timer(10, self._xeng_qdr_okay, sensor).start()
+        return
+
+    def _feng_qdr_okay(self, sensor):
+        """
+        f-engine QDR check
+        :param sensor:
+        :return:
+        """
+        host_name = sensor.name.split('_')[2]
+        result = self.fhosts[host_name].qdr_okay()
+        sensor.set(time.time(), Sensor.NOMINAL if result else Sensor.ERROR, result)
+        self.logger.info('_feng_qdr_okay ran on %s' % host_name)
+        IOLoop.instance().call_later(self._feng_qdr_okay, 10)
```

Comment:
use self.ioloop (or at least IOLoop.current()), since there is no guarantee that IOLoop.instance() will be the "correct" ioloop for us. 

IOLoop.current() will be correct if you used ioloop.add_callback(), since this method will then be called in "our" ioloop's thread context, hence IOLoop.current() will definitely give the correct IOLoop instance.